### ISSUE TITLE
Format fails when saving file

### ISSUE THREAD

**Issue by my-vegetable-has-exploded** [microsoft/vscode-cpptools#5898]
**Type: LanguageService**

**Describe the bug** 
- OS and Version: Windows 1909
- VS Code Version:  1.47.3
- C/C++ Extension Version: 0.29.0  
- Other extensions you installed (and if the issue persists after disabling them):
- Does this issue involve using SSH remote to run the extension on a remote machine?: No
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

**Steps to reproduce**

1. use ctrl+s to format at save
2. formating fails， and some characters has been deleted.

**Expected behavior**

<details>
  <summary><strong>Logs</strong></summary>

```
Insert logs here.
```
</details>

**Screenshots**

**Additional context**

**Comment by sean-mcmanus** [comment]
Are you sure you have 0.29.0 and not 0.30.0-insiders?

Does formatting fail with Format Document instead of format on save?

Are you able to provide a repro code sample and a list of edit steps to make that can cause a repro?

We need more info on your formatting settings. With 0.30.0-insders, a new vcFormat value for C_Cpp.formatting uses a new formatting implementation, which is also used if the C_Cpp.clang_format_style is set to Visual Studio (or if the C_Cpp.clang_format_fallback_style is used due to no .clang-format file being found). Is the issue resolved if you change the styles to use `Emulated Visual Studio`?

**Comment by my-vegatable-has-exploded** [comment]
Sorry, it occurs in 0.30-insiders.

Both two kinds of format fail.

But format still fails after i changed setting as your suggestion. One of my classmate told me that he resets all extension settings and it works finally. 

There isn't clang environment in my computer.

What Detailed information needs to be included in the repro code？

```json
"C_Cpp.clang_format_style": "Visual Studio",
"C_Cpp.formatting": "vcFormat"
```

**Comment by my-vegetable-has-exploded** [comment]
> Are you sure you have 0.29.0 and not 0.30.0-insiders?

> Sorry, it occurs in 0.30-insiders.

> Does formatting fail with Format Document instead of format on save?

> Both two kinds of formatting fail.

> Are you able to provide a repro code sample and a list of edit steps to make that can cause a repro?

>What Detailed information needs to be included in the repro code？

```json
"C_Cpp.clang_format_style": "Visual Studio",
"C_Cpp.formatting": "vcFormat"
```
> We need more info on your formatting settings. With 0.30.0-insders, a new vcFormat value for C_Cpp.formatting uses a new formatting implementation, which is also used if the C_Cpp.clang_format_style is set to Visual Studio (or if the C_Cpp.clang_format_fallback_style is used due to no .clang-format file being found). Is the issue resolved if you change the styles to use `Emulated Visual Studio`?

>Formatting still fails after i changed setting as your suggestion. One of my classmate told me that he resets all extension settings and it works finally. 

>There isn't clang environment in my computer.

**Comment by sean-mcmanus** [comment]
Does the issue repro if you change "C_Cpp.formatting" to "clangFormat", "C_Cpp.clang_format_style" to "Emulated Visual Studio", and "C_Cpp.clang_format_fallbackStyle" to "Emulated Visual Studio"? Our extension has clang-format bundled, so it will use that if you don't have it installed globally.

So far we haven't been able to witness document corruption. So if we could get some repro like
1. Paste code "...".
2. Make edits...
3. Format document.
then that would help us to debug and fix the issue.

Also, did you change you any C_Cpp.vcFormat.* settings?

**Comment by sean-mcmanus** [comment]
We got a repro at https://github.com/microsoft/vscode-cpptools/issues/5914, so I don't think we need any more info. We'll try to get a fix soon (the workaround mentioned previously should work in the meantime).

**Comment by my-vegetable-has-exploded** [comment]
> We got a repro at #5914, so I don't think we need any more info. We'll try to get a fix soon (the workaround mentioned previously should work in the meantime).

Thanks，I will keep using 0.29 these days.

**Comment by sean-mcmanus** [comment]
This is fixed with 0.30.0-insiders2: https://github.com/microsoft/vscode-cpptools/releases/tag/0.30.0-insiders2 .
